src/components/victory-line.jsx

```diff
@@ -89,6 +90,11 @@ export default class VictoryLine extends React.Component {
      */
     label: React.PropTypes.string,
     /**
+     * The labelComponent prop takes in an entire, HTML-complete label component
+     * which will be used to create labels for line to use
+     */
+    labelComponent: React.PropTypes.element,
+    /**
```

Comment:
Why not just one prop, `label`? Having two props is confusing IMO, as the behavior if I specify both is unclear. If we get a string for `label`, then `getLabel` can just wrap it in `<VictoryLabel>` right off the bat and now you're always dealing with an element – and have a lot fewer `&&`'s and `||`'s.

If we kept the `labelComponent` class, I also wouldn't expect it to be an element instance, but a class (or string, if you want a builtin HTML element). Like `labelComponent={MyLabelClass}` or `labelComponent="div"`. This would mirror [how `ReactTransitionGroup` does it](https://facebook.github.io/react/docs/animation.html#rendering-a-different-component).